### PR TITLE
Issue #3935 - As a funder, I want permission / reserved for UI on bounty new to be changed, so I can have a clean UX

### PR DESCRIPTION
**Pull request by francopetra**
##### Description

- Added conditional to show "Reserved for" in the Permissions dropdown only when a funder selects the Project Type: Traditional (not for Cooperative or Contest).
- Added Reserved to filters in the Issue Explorer, Tooltips and Issue Details.

Fixes: https://github.com/gitcoinco/web/issues/3935
Refs: https://gitcoin.co/issue/gitcoinco/web/3935/2569

Project Type: Traditional (No user selected)   
Screen Shot 2019-03-24 at 12 49 02 AM

Project Type: Traditional (User selected)  
Screen Shot 2019-03-24 at 1 34 08 AM

Project Type: Contest   
Screen Shot 2019-03-24 at 12 49 12 AM

Issue Explorer filter
Screen Shot 2019-03-24 at 12 52 07 AM

##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [ ] Ran `make test` and everything passed! 

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by francopetra** [comment]
Hi Gitcoin team! Just to be sure, I haven't rebased in a PR from the UI before, but as far as I understand it can be done after the app is reviewed right?
Would appreciate indications if it has to be done any other way. 
Also looking forward to the code review!

File: app/dashboard/models.py [hunk 1]
```diff
@@ -176,6 +176,7 @@ class Bounty(SuperModel):
     PERMISSION_TYPES = [
         ('permissionless', 'permissionless'),
         ('approval', 'approval'),
+        ('reserved', 'reserved'),
```
Comment (owocki) [inline app/dashboard/models.py]: should we migrate existing bounties who are reserved to the new permission type?

**Comment by owocki** [comment]
thanks @francopetra !  exciting to see the progress here.  how did you test?  specifically i'd be curious to hear if

- legacy bounties work okay
- bounty details page looks okay
- can the person who's reserved for, click 'start work'?
- can the person who's NOT reserved for, click 'start work'?

**Comment by francopetra** [comment]
Thanks for start reviewing my work @owocki . 
The way I tested was creating another test account on GitHub and used an Eth account for each GitHub account. 
So I checked the cases manually on the Rinkeby network.
The only difference from what I understand is that it doesn't say "Start work", instead it says "Express Interest", but as soon as the person press the button and fills out the why s/he is interested they look as approved, and since they are the only ones who can do this it's hard to go any other way.
Will share some screenshots so we are on the same page:

**Francopretra funds Issue and reserve it for PixelantDesign**
Screen Shot 2019-03-25 at 11 53 58 PM

**Testeandot tries to look into the issue, but there's no button to start working or express interest**
Screen Shot 2019-03-26 at 12 10 35 AM

**Francopretra funds Issue and reserve it for Testeandot**
Screen Shot 2019-03-26 at 12 22 21 AM

**Testeandot has option "Express Interest". After completing the pop-up s/he can start working**
Screen Shot 2019-03-26 at 12 23 35 AM

Screen Shot 2019-03-26 at 12 26 27 AM

After page is refreshed the whole UI updates with SubmitWork button and an updated activity.
Screen Shot 2019-03-26 at 12 37 14 AM

**Comment by SaptakS** [comment]
@francopetra can you please resolve the conflicts?

**Comment by francopetra** [comment]
> @francopetra can you please resolve the conflicts?

Done! Let me know if there's anything else I can help with.

File: app/assets/v2/js/pages/new_bounty.js [hunk 2]
```diff
@@ -131,45 +134,26 @@ $(document).ready(function() {
...
 
-  $('select[name=denomination]').change(function(e) {
-    const token = tokenAddressToDetails(e.target.value).name;
+  var option = gettext('Reserved For - I will select a Gitcoin user.');
 
-    $('#summary-bounty-token').html(token);
-    $('#summary-fee-token').html(token);
-    populateBountyTotal();
-  });
+  $('select[name=project_type]').on('change', function() {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/new_bounty.js]: ❤️

Comment (francopetra) [reply app/assets/v2/js/pages/new_bounty.js]: Hey @thelostone-mc . Didn't see these conflicts. How do we work them out?

File: app/assets/v2/js/pages/new_bounty.js [hunk 3]
```diff
@@ -131,45 +134,26 @@ $(document).ready(function() {
...
+    var val = $('select[name=project_type] option:selected').val();
 
-  $('#featuredBounty').on('change', function() {
```
Comment (SaptakS) [inline app/assets/v2/js/pages/new_bounty.js]: Why did you delete the featuredBounty code?

File: app/dashboard/models.py (hunk 1 above)
Comment (SaptakS) [reply app/dashboard/models.py]: @owocki what would you say is the best effective way to do this? I would like to just put relevant code in migration file, but we can also write a management command to do the same. @francopetra @thelostone-mc

Comment (owocki) [reply app/dashboard/models.py]: write a management command and/or a migration to do it.  migration will be run automatically, so id prefer that.  if mgmt command, we just need to remind @danlipert to run it in prod after he deploys

Comment (danlipert) [reply app/dashboard/models.py]: @francopetra @owocki @SaptakS A new migration would be ideal :+1:

Comment (francopetra) [reply app/dashboard/models.py]: @danlipert @owocki Hey, as I understand I should delete this, and already created a migration in app/dashboard/migrations/0001_initial.py (Line 94). 
Not sure how to migrate existing ones though.

File: app/dashboard/migrations/0001_initial.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ class Migration(migrations.Migration):
                 ('canceled_on', models.DateTimeField(blank=True, null=True)),
                 ('canceled_bounty_reason', models.TextField(blank=True, default='', verbose_name='Cancelation reason')),
                 ('project_type', models.CharField(choices=[('traditional', 'traditional'), ('contest', 'contest'), ('cooperative', 'cooperative')], default='traditional', max_length=50)),
-                ('permission_type', models.CharField(choices=[('permissionless', 'permissionless'), ('approval', 'approval')], default='permissionless', max_length=50)),
+                ('permission_type', models.CharField(choices=[('permissionless', 'permissionless'), ('approval', 'approval'), ('reserved', 'reserved'),], default='permissionless', max_length=50)),
```
Comment (danlipert) [inline app/dashboard/migrations/0001_initial.py]: I think we need a new migration here instead of editing the old one, right?

File: app/dashboard/models.py (hunk 1 above)
Comment (danlipert) [reply app/dashboard/models.py]: @francopetra we're talking a bout a new migration as described here: https://docs.djangoproject.com/en/2.2/topics/migrations/ with a custom forwards method: https://docs.djangoproject.com/en/2.2/howto/writing-migrations/

**Comment by thelostone-mc** [comment]
Closing this out as the PR has gone stale! Feel free to open up a fresh PR and with all the comments addressed we can get stuff in
